File: javascript/src/problems/two-sum.js
/**
 * 1. Two Sum [Easy] https://leetcode.com/problems/two-sum/
 * 
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
    const numToIndex = new Map();
    for (let i = 0; i < nums.length; i++) {
        let numToLookUp = target - nums[i];
        if (numToIndex.has(numToLookUp)) {
            return [numToIndex.get(numToLookUp), i];
        } else {
            numToIndex.set(nums[i], i);   
        }        
    }
    return [];
};

module.exports = { twoSum };

File: java/src/main/java/org/carlosbello/leetcode/problems/TwoSum.java
package org.carlosbello.leetcode.problems;

import java.util.HashMap;
import java.util.Map;

/**
 * 1. Two Sum [easy] https://leetcode.com/problems/two-sum/
 */
public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexByNum = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int compliment = target - nums[i];
            if (indexByNum.containsKey(compliment)) {
                return new int[] {i, indexByNum.get(compliment)};
            } else {
                indexByNum.put(nums[i], i);
            }
        }
        return new int[0];
    }
}
